projectview/src/test/java/org/jetbrains/bsp/bazel/projectview/parser/sections/ProjectViewTargetsSectionParserTest.java

```diff
@@ -0,0 +1,217 @@
+  }
+
+  // ProjectViewTargetsSection parse(rawSection)
+
+  @Test
+  public void shouldThrowIllegalArgumentExceptionForWrongSectionName() {
+    // given
+    var rawSection = new ProjectViewRawSection("wrongsection", "-bodyelement");
+
+    // when
+    var sectionTry = parser.parse(rawSection);
+
+    // then
+    assertTrue(sectionTry.isFailure());
+
+    assertEquals(
+        "Project view parsing failed! Expected 'targets' section name, got 'wrongsection'!",
+        sectionTry.getCause().getMessage());
+  }
+
+  @Test
+  public void shouldParseEmptySectionBody() {
+    // given
+    var rawSection = new ProjectViewRawSection("targets", "");
+
+    // when
+    var sectionTry = parser.parse(rawSection);
+
+    // then
+    assertTrue(sectionTry.isSuccess());
+    var section = sectionTry.get();
+
+    var expectedSection = new ProjectViewTargetsSection(List.of(), List.of());
+    assertEquals(expectedSection, section);
+  }
+
+  @Test
+  public void shouldParseIncludedTargets() {
+    // given
+    var sectionBody =
+        "  //test_included1:test1\n\t//:test_included1:test2\n//:test_included2:test1\n\n";
+    var rawSection = new ProjectViewRawSection("targets", sectionBody);
+
+    // when
+    var sectionTry = parser.parse(rawSection);
+
+    // then
+    assertTrue(sectionTry.isSuccess());
+    var section = sectionTry.get();
+
+    var expectedSection =
+        new ProjectViewTargetsSection(
+            List.of("//test_included1:test1", "//:test_included1:test2", "//:test_included2:test1"),
+            List.of());
+    assertEquals(expectedSection, section);
+  }
+
+  @Test
+  public void shouldParseExcludedTargets() {
+    // given
+    var sectionBody =
+        "  -//test_excluded1:test1\n\t-//test_excluded1:test2\n-//test_excluded2:test1\n\n";
+    var rawSection = new ProjectViewRawSection("targets", sectionBody);
+
+    // when
+    var sectionTry = parser.parse(rawSection);
+
+    // then
+    assertTrue(sectionTry.isSuccess());
+    var section = sectionTry.get();
+
+    var expectedSection =
+        new ProjectViewTargetsSection(
+            List.of(),
+            List.of("//test_excluded1:test1", "//test_excluded1:test2", "//test_excluded2:test1"));
+    assertEquals(expectedSection, section);
+  }
+
+  @Test
+  public void shouldParseIncludedAndExcludedTargets() {
+    // given
+    var sectionBody =
+        "  -//test_excluded1:test1\n\t//test_included1:test1\n-//test_excluded1:test2\n\n";
+    var rawSection = new ProjectViewRawSection("targets", sectionBody);
+
+    // when
+    var sectionTry = parser.parse(rawSection);
+
+    // then
+    assertTrue(sectionTry.isSuccess());
+    var section = sectionTry.get();
+
+    var expectedSection =
+        new ProjectViewTargetsSection(
+            List.of("//test_included1:test1"),
+            List.of("//test_excluded1:test1", "//test_excluded1:test2"));
+    assertEquals(expectedSection, section);
+  }
+
+  // ProjectViewTargetsSection parse(rawSections)
```

Comment:
Maybe a bit more descriptive, so it immediately stands out how these tests differ from the ones that parse a single section.

Reply:
I want to bump JUnit to version 5, then I will split it into nested tests, with better description